nodestream/extractors/stores/aws/s3_extractor.py

```diff
@@ -14,30 +14,45 @@ def from_file_data(
         bucket: str,
         prefix: Optional[str] = None,
         object_format: Optional[str] = None,
+        archive_dir: str = None,
         **aws_client_args,
     ):
         return cls(
             bucket=bucket,
             object_format=object_format,
             prefix=prefix,
+            archive_dir=archive_dir,
             s3_client=AwsClientFactory(**aws_client_args).make_client("s3"),
         )
 
     def __init__(
         self,
         bucket: str,
+        archive_dir: str,
```

Comment:
```suggestion
        archive_dir:  Optional[str],
```
Same thing as above.

Reply:
Same.  Done, please double check me